api/api.yaml

```diff
@@ -0,0 +1,361 @@
+        5XX:
+          description: Unexpected error.
+      security: null
+    put:
+      tags:
+        - User
+      summary: Update an User
+      parameters:
+        - in: path
+          name: userId
+          description: User ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/User"
+      responses:
+        "201":
+          description: Created
+  /user:
+    post:
+      tags:
+        - User
+      summary: Add a new User
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/User"
+      responses:
+        "201":
+          description: Created
+  /session/{sessionId}:
+    get:
+      tags:
+        - Session
+      summary: Gets Session by id
+      description: >
+        This resource represents an individual Session in the system. Each
+        Session is identified by a numeric `id`.
+      operationId: findSessionById
+      parameters:
+        - in: path
+          name: sessionId
+          description: Session ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/User"
+        "400":
+          description: Invalid status value
+      security: null
+  /tags/{subjectID}:
+    get:
+      tags:
+        - Tags
+      summary: Gets all Tags by the Subject ID
+      operationId: findTagsBySubjectID
+      parameters:
+        - in: path
+          name: subjectID
+          description: Subject ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        "200":
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Tag"
+        "400":
+          description: Bad request. Subject ID must be an integer and larger than 0.
+        "401":
+          description: Authorization information is missing or invalid.
+        "404":
+          description: An subject with the specified ID was not found.
+        5XX:
+          description: Unexpected error.
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        username:
+          type: string
+        email:
+          type: string
+        lastLogin:
+          type: string
+        semester:
+          type: integer
+        course:
+          type: string
+          description: Zahnmedizin or Humanmedizin
```

Comment:
I'd change this to int or enum. e.g. Humanmed.: 0 (so it's standard), Zahmed.: 1, both: 2